hedera-mirror-rosetta/app/services/construction/construction_service_test.go

```
@@ -0,0 +1,461 @@
+
+	// then:
+	assert.Equal(t, expectedConstructionParseResponse, res)
+	assert.Nil(t, e)
+}
+
+func TestConstructionParseThrowsWhenDecodeStringFails(t *testing.T) {
+	// when:
+	res, e := NewConstructionAPIService().ConstructionParse(nil, dummyConstructionParseRequest(invalidTxHexStr, false))
+
+	// then:
+	assert.Nil(t, res)
+	assert.Equal(t, errors.Errors[errors.TransactionDecodeFailed], e)
+}
+
+func TestConstructionParseThrowsWhenUnmarshallFails(t *testing.T) {
+	// when:
+	res, e := NewConstructionAPIService().ConstructionParse(nil, dummyConstructionParseRequest(corruptedTxHexStr, false))
+
+	// then:
+	assert.Nil(t, res)
+	assert.Equal(t, errors.Errors[errors.TransactionUnmarshallingFailed], e)
+}
+
+func TestConstructionPayloads(t *testing.T) {
+	// given:
+	var expectedNilBytes []byte
+	operations := []*types.Operation{
+		dummyOperation(0, "CRYPTOTRANSFER", "0.0.123352", "-1000"),
+		dummyOperation(1, "CRYPTOTRANSFER", "0.0.123518", "1000"),
+	}
+	expectedPayloadsResponse := &types.ConstructionPayloadsResponse{
+		UnsignedTransaction: "0x1a00223d0a140a0c08faf3b5fc0510c681d7d303120418e3cc13120218061880c2d72f2202087872180a160a090a0418b9c30710c8010a090a0418e3cc1310c701",
+		Payloads: []*types.SigningPayload{
+			{
+				AccountIdentifier: &types.AccountIdentifier{
+					Address: "0.0.123352",
+				},
+				Bytes: expectedNilBytes,
+			},
+		},
+	}
+
+	// when:
+	res, e := NewConstructionAPIService().ConstructionPayloads(nil, dummyPayloadsRequest(operations))
+
+	// then:
+	// here we do not assert the whole response object to equal the expected one, because invocation of this method appends a unique timestamp to the result, thus making the signed TX and Bytes unique and non-assertable.
+	assert.Equal(t, expectedPayloadsResponse.Payloads[0].AccountIdentifier.Address, res.Payloads[0].AccountIdentifier.Address)
+	assert.Nil(t, e)
+}
+
+func TestConstructionPayloadsThrowsWhenInvalidOperationsSum(t *testing.T) {
```

it's about the implementation of `ConstructionPayloads` not the test case:

should `ConstructionPayloads` fail early when supplied with an empty `Operations` slice?